src/StoreApi/Utilities/CartController.php

```diff
@@ -388,92 +453,54 @@ public function validate_cart_items() {
 			}
 		}
 
-		$error = new WP_Error();
-
-		if ( count( $out_of_stock_products ) > 0 ) {
-			$singular_error = $this->get_error_message_for_stock_exception_type( 'out_of_stock', 'singular' );
-			$plural_error   = $this->get_error_message_for_stock_exception_type( 'out_of_stock', 'plural' );
-
-			$error->add(
-				409,
-				$this->add_product_names_to_message( $singular_error, $plural_error, $out_of_stock_products )
-			);
-		}
-
-		if ( count( $not_purchasable_products ) > 0 ) {
-			$singular_error = $this->get_error_message_for_stock_exception_type( 'not_purchasable', 'singular' );
-			$plural_error   = $this->get_error_message_for_stock_exception_type( 'not_purchasable', 'plural' );
-
-			$error->add(
-				409,
-				$this->add_product_names_to_message( $singular_error, $plural_error, $not_purchasable_products )
-			);
-		}
-
-		if ( count( $too_many_in_cart_products ) > 0 ) {
-			$singular_error = $this->get_error_message_for_stock_exception_type( 'too_many_in_cart', 'singular' );
-			$plural_error   = $this->get_error_message_for_stock_exception_type( 'too_many_in_cart', 'plural' );
-
-			$error->add(
-				409,
-				$this->add_product_names_to_message( $singular_error, $plural_error, $too_many_in_cart_products )
-			);
-		}
+		if ( count( $errors ) > 0 ) {
 
-		if ( count( $partial_out_of_stock_products ) > 0 ) {
-			$singular_error = $this->get_error_message_for_stock_exception_type( 'partial_out_of_stock', 'singular' );
-			$plural_error   = $this->get_error_message_for_stock_exception_type( 'partial_out_of_stock', 'plural' );
+			$error = new WP_Error();
```

Comment:
Might this be useful here? https://developer.wordpress.org/reference/classes/wp_error/merge_from/

Reply:
It won't make _much_ of a difference because we have a `WP_Error[]` type. But it's a lot _cleaner_! :)